plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/InternalAwsS3Service.java

```diff
@@ -166,10 +166,10 @@ protected void doClose() throws ElasticsearchException {
     }
 
     static class PrivilegedInstanceProfileCredentialsProvider implements AWSCredentialsProvider {
-        private final InstanceProfileCredentialsProvider credentials;
+        private final DefaultAWSCredentialsProviderChain credentials;
 
         private PrivilegedInstanceProfileCredentialsProvider() {
-            this.credentials = new InstanceProfileCredentialsProvider();
+            this.credentials = new DefaultAWSCredentialsProviderChain();
```

Comment:
This is incorrect. This needs to be turned into a chain over InstanceProfile provider and container credentials provider as I described in https://github.com/elastic/elasticsearch/issues/23039#issuecomment-279503484. We will not use the default chain because it exposes insecure methods of passing credentials.